projects/gateway2/query/backendmap.go

```diff
@@ -25,7 +25,7 @@ func backendToRefKey(ref gwv1.BackendObjectReference) backendRefKey {
 	const delim = "~"
 	return backendRefKey(
 		string(ptrOrDefault(ref.Group, "")) + delim +
-			string(ptrOrDefault(ref.Kind, "")) + delim +
+			string(ptrOrDefault(ref.Kind, "Service")) + delim +
```

Comment:
What impact does this have? I'm not sure I understand it enough

Reply:
This will include "Service" when the provided backendRef does not include the `kind` field. This change properly implements the Gateway API spec ([xref](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/object_reference_types.go#L120)). When calling methods for this type, i.e. `get(backendRef)`, the provided backendRef.kind may be nil in which "Service" should be used.